NitroxPatcher/Patches/Dynamic/Constructor_Deploy_Patch.cs

```diff
@@ -0,0 +1,32 @@
+using System.Reflection;
+using HarmonyLib;
+using NitroxClient.GameLogic;
+using NitroxClient.MonoBehaviours;
+using NitroxModel.Core;
+using NitroxModel.DataStructures;
+using NitroxModel.Helper;
+
+namespace NitroxPatcher.Patches.Dynamic;
+
+public class Constructor_Deploy_Patch : NitroxPatch, IDynamicPatch
+{
+    public static readonly MethodInfo TARGET_METHOD = Reflect.Method((Constructor t) => t.Deploy(default(bool)));
+
+    public static void Prefix(Constructor __instance, bool value)
+    {
+        // only trigger updates when there is a valid state change.
+        if(value != __instance.deployed)
+        {
+            NitroxId id = NitroxEntity.GetId(__instance.gameObject);
+            Entities entities = NitroxServiceLocator.LocateService<Entities>();
+            entities.EntityMetadataChanged(__instance, id);
```

Comment:
```suggestion
            Resolve<Entities>().EntityMetadataChanged(__instance, id);
```

Reply:
Note: We're now using Resolve which also uses NitroxServiceLocator, but it's providing a cache

Reply:
Thanks this is better.